examples/MomentoApplication/Program.cs

```diff
@@ -6,43 +6,94 @@
+// Use a GUID for the cache name so it's unlikely to exist already.
+string CACHE_NAME = Guid.NewGuid().ToString();
+const string KEY = "MyKey";
+const string VALUE = "MyData";
+const uint DEFAULT_TTL_SECONDS = 60;
+
+using (SimpleCacheClient client = new SimpleCacheClient(Configurations.Laptop.Latest, MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS))
 {
-    class Program
+    Console.WriteLine($"Creating cache {CACHE_NAME}");
+    var createCacheResponse = await client.CreateCacheAsync(CACHE_NAME);
+    // Check the create response for an error and handle as appropriate.
+    if (createCacheResponse is CreateCacheResponse.Error createError)
     {
-        static readonly String MOMENTO_AUTH_TOKEN = Environment.GetEnvironmentVariable("MOMENTO_AUTH_TOKEN");
-        static readonly String CACHE_NAME = "cache";
-        static readonly String KEY = "MyKey";
-        static readonly String VALUE = "MyData";
-        static readonly uint DEFAULT_TTL_SECONDS = 60;
-
-        async static Task Main(string[] args)
+        if (createError.ErrorCode == MomentoErrorCode.LIMIT_EXCEEDED_ERROR)
+        {
+            Console.WriteLine("Error: cache limit exceeded. We need to talk to support@moentohq.com! Exiting.");
+        } else
         {
-            using SimpleCacheClient client = new SimpleCacheClient(Configurations.Laptop.Latest, MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS);
-            var createCacheResult = await client.CreateCacheAsync(CACHE_NAME);
+            Console.WriteLine($"Error creating cache: {createError.Message}. Exiting.");
+        }
+        // Any error is considered fatal.
+        Environment.Exit(1);
+    }
+    // If there's already a cache by this name, we don't want to modify it.
+    if (createCacheResponse is CreateCacheResponse.CacheAlreadyExists)
+    {
+        Console.WriteLine($"Error: strangely, a cache with the name {CACHE_NAME} already exists. Exiting.");
+        Environment.Exit(1);
+    }
 
-            Console.WriteLine("Listing caches:");
-            String token = null;
-            do
-            {
-                ListCachesResponse resp = await client.ListCachesAsync(token);
-                if (resp is ListCachesResponse.Success successResult)
-                {
-                    foreach (CacheInfo cacheInfo in successResult.Caches)
-                    {
-                        Console.WriteLine(cacheInfo.Name);
-                    }
-                    token = successResult.NextPageToken;
-                }
-            } while (!String.IsNullOrEmpty(token));
-            Console.WriteLine($"\nSetting key: {KEY} with value: {VALUE}");
-            await client.SetAsync(CACHE_NAME, KEY, VALUE);
-            Console.WriteLine($"\nGet value for  key: {KEY}");
-            CacheGetResponse getResponse = await client.GetAsync(CACHE_NAME, KEY);
-            if (getResponse is CacheGetResponse.Hit hitResponse)
+    Console.WriteLine("\nListing caches:");
+    String token = null;
```

Comment:
What are the differences between `String` and `string` in C#?

Reply:
Essentially there is none. `string` is an alias for `System.String`.